frontend/public/components/storage/attach-storage.tsx

```diff
@@ -160,186 +170,160 @@ class AttachStorageForm extends React.Component<
       });
     });
 
-    this.setState({ inProgress: true });
-    this.createPVCIfNecessary().then(claimName => {
+    setInProgress(true);
+    createPVCIfNecessary().then(pvClaimName => {
       // add new volume to the pod template if not preent
       if (!volumeAlreadyMounted) {
         template.spec.volumes = template.spec.volumes || [];
         template.spec.volumes.push({
           name: volumeName,
           persistentVolumeClaim: {
-            claimName,
+            claimName: pvClaimName,
           },
         });
       }
-      k8sUpdate(kindObj, resourceObj, metadata.namespace, metadata.name).then(
+
+      k8sUpdate(kindObj, updatedResourceObj, namespace, resourceName).then(
         resource => {
-          this.setState({ inProgress: false });
+          setInProgress(false);
           history.push(resourceObjPath(resource, referenceFor(resource)));
         },
-        err => this.setState({ error: err.message, inProgress: false })
-      );
-    }, err => this.setState({ error: err.message, inProgress: false }));
-
+        err => {
```

Comment:
I don't believe you need this twice.... You should be able to have one outer `catch`